container/docker/factory.go

```diff
@@ -204,6 +208,35 @@ func parseDockerVersion(full_version_string string) ([]int, error) {
 	return version_array, nil
 }
 
+func getStorageDir(dockerInfo *docker.Env) (string, error) {
+	storageDriverInfo := dockerInfo.GetList("DriverStatus")
+	if len(storageDriverInfo) == 0 {
+		return "", fmt.Errorf("failed to find docker storage driver options")
+	}
+
+	entries := map[string]string{}
+	for _, data := range storageDriverInfo {
+		if data == "" {
+			continue
+		}
+		var parts [][]string
+		if err := json.Unmarshal([]byte(data), &parts); err != nil {
+			return "", fmt.Errorf("failed to parse docker storage driver options - %+v", data)
+		}
+		for _, part := range parts {
+			if len(part) != 2 {
```

Comment:
You are probably right, I am not that familiar with the string format that docker provides or what the go deserialization libraries are capable of.  I was hoping the deserialization libraries would coerce a slice of slices len=2 into a map.